docs_espressif/en/additionalfeatures/cmakelists-editor.rst

```diff
@@ -0,0 +1,56 @@
+CMakeLists.txt Editor
+==============================
+
+When you right click on any CMakeLists.txt file this extension provides a custom CMakeLists.txt Editor to fill an ESP-IDF Project and Component Registration as specified in:
+
+- `ESP-IDF Project CMakeLists.txt <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/build-system.html#project-cmakelists-file>`_
+- `ESP-IDF Component CMakeLists.txt Files <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/build-system.html#component-cmakelists-files>`_
+
+You need to choose which kind of CMakeLists.txt file (project or component) to edit. There is 2 types of input, one is a simple string and another is an array of strings, such as Component Sources (SRCS).
+
+.. note::
+  * All inputs are described in the `CMakeLists.txt schema <https://github.com/espressif/vscode-esp-idf-extension/blob/master/cmakeListsSchema.json>`_
+  * This editor doesn't support all CMake functions and syntaxes. This editor should only be used for simple CMakeLists.txt options such as component registration (using idf_component_register) and basic project elements. If you need more customization or advanced CMakeLists.txt, consider reviewing `ESP-IDF Build System <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/build-system.html>`_. Also review **CMakeLists.txt editor schema** for a list of supported code.
+  * THIS WILL OVERRIDE ANY EXISTING CODE IN THE FILE WITH THE ONE GENERATED IN THE EDITOR. IF YOU HAVE ANY CODE NOT INCLUDED IN THE SCHEMA (OR SINGLE LINE COMMENTS) USE A REGULAR TEXT EDITOR INSTEAD**
```

Comment:
I don't think we should use caps to get user's attention, but maybe we can use this type of styling and put it at the top of this article
<img width="754" alt="Screenshot 2024-08-08 at 17 02 18" src="https://github.com/user-attachments/assets/5876554c-e6bc-4b5c-8bf2-a5e591a3c76a">

Reply:
or 
<img width="754" alt="Screenshot 2024-08-08 at 17 15 47" src="https://github.com/user-attachments/assets/029edd70-82dd-4e7f-9e56-13414b89bac5">